plugins/SLURM-Integration/ufm_slurm_utils.py

```diff
@@ -187,28 +191,32 @@ class UFM:
     def getUrl(self, resource_path, auth_type=Constants.BASIC_AUTH):
         if auth_type == Constants.TOKEN_AUTH:
             url = resource_path.replace("ufmRest", "ufmRestV3")
+        elif auth_type == Constants.KERBEROS_AUTH:
+            url = resource_path.replace("ufmRest", "ufmRestKrb")
         else:
             url = resource_path
         return url
 
-    def getServerSession(self, auth_type=None, username=None, password=None, token=None):
+    def getServerSession(self, auth_type=None, username=None, password=None, token=None, principal_name=None,
+                         principal_password=""):
         """
         Creating REST client session for server connection,
         after globally setting Authorization,
         Content-Type and charset for session.
         """
         session = requests.Session()
+        session.verify = False
+        session.headers.update({'Content-Type': 'application/json; charset=utf-8'})
 
         if auth_type == Constants.BASIC_AUTH:
             session.auth = (username, password)
-            session.verify = False
-            session.headers.update(
-                {'Content-Type': 'application/json; charset=utf-8'})
         elif auth_type == Constants.TOKEN_AUTH:
-            session.verify = False
-            session.headers.update(
-                {'Content-Type': 'application/json; charset=utf-8',
-                 'Authorization': 'Basic %s' % token})
+            session.headers.update({'Authorization': 'Basic %s' % token})
+        elif auth_type == Constants.KERBEROS_AUTH:
+            __, krb_context = kerberos.authGSSClientInit("HTTP", principal_name)
+            kerberos.authGSSClientStep(krb_context, principal_password)
```

Comment:
with user principal, did you try to authenticate without adding the password in the config file, considering that the ticket was already generated?

Reply:
I tried it without sending the password and saw that it's working without any issues! because we are using principal name only with kerberos authentication after generating the TGT, So the password is redundant I will remove it form the ufm_slurm.conf file.